packages/frontend/@n8n/design-system/src/v2/components/DropdownMenu/DropdownMenu.vue

```diff
@@ -0,0 +1,355 @@
+
+		<component :is="teleported ? DropdownMenuPortal : 'template'">
+			<DropdownMenuContent
+				:id="id"
+				ref="contentRef"
+				:class="[$style.content, extraPopperClass]"
+				:side="placementParts.side"
+				:align="placementParts.align"
+				:side-offset="5"
+				:style="contentStyle"
+				@keydown="handleContentKeydown"
+			>
+				<slot v-if="slots.content" name="content" />
+				<template v-else>
+					<N8nDropdownMenuSearch
+						v-if="searchable"
+						ref="searchRef"
+						:model-value="searchTerm"
+						:placeholder="searchPlaceholder"
+						:show-icon="showSearchIcon"
+						@update:model-value="handleSearchUpdate"
+						@key:escape="close"
+						@key:navigate="navigation.navigate"
+						@key:arrow-right="navigation.handleArrowRight"
+						@key:arrow-left="navigation.handleArrowLeft"
+						@key:enter="navigation.handleEnter"
+					/>
+
+					<template v-if="loading">
+						<slot name="loading">
+							<div :class="$style['loading-container']">
+								<N8nLoading
+									v-for="i in loadingItemCount"
+									:key="i"
+									:rows="1"
+									:class="$style['loading-item']"
+									variant="p"
+								/>
+							</div>
+						</slot>
+					</template>
+					<template v-else-if="items.length === 0">
+						<slot name="empty">
+							<div :class="$style['empty-state']">No items</div>
+						</slot>
+					</template>
+					<template v-else>
+						<template v-for="(item, index) in items" :key="item.id">
+							<slot name="item" :item="item">
+								<N8nDropdownMenuItem
+									v-bind="item"
+									:highlighted="highlightedIndex === index"
+									:sub-menu-open="openSubMenuIndex === index"
+									@select="handleItemSelect"
+									@search="handleItemSearch"
+									@update:sub-menu-open="(open: boolean) => handleSubMenuOpenChange(index, open)"
+								>
+									<template v-if="slots['item-leading']" #item-leading="{ ui }">
+										<slot name="item-leading" :item="item" :ui="ui" />
+									</template>
+									<template v-if="slots['item-trailing']" #item-trailing="{ ui }">
+										<slot name="item-trailing" :item="item" :ui="ui" />
+									</template>
+								</N8nDropdownMenuItem>
+							</slot>
+						</template>
+					</template>
+				</template>
+			</DropdownMenuContent>
+		</component>
+	</DropdownMenuRoot>
+</template>
+
+<style module>
+.activator {
+	display: inline-flex;
+	align-items: center;
+	justify-content: center;
+	width: var(--spacing--xl);
+	height: var(--spacing--xl);
+	padding: 0;
+	border: none;
+	border-radius: var(--radius);
+	background-color: transparent;
+	color: var(--color--text);
+	cursor: pointer;
+	outline: none;
+
+	&:hover {
+		background-color: var(--color--background);
+	}
+
+	&:focus-visible {
+		box-shadow: 0 0 0 var(--spacing--5xs) var(--color--primary);
+	}
+
+	&[data-disabled] {
+		color: var(--color--text--tint-1);
+		cursor: not-allowed;
+	}
+}
+
+.activator-emoji {
+	font-size: var(--font-size--md);
+	line-height: 1;
+}
+
+.content {
```

Comment:
The wrapper for the `MenuItems` needs padding of 4px all the way around. Not just on the top.

There should also be a 2px gap between each menut item

<img width="196" height="360" alt="Image" src="https://github.com/user-attachments/assets/86033aba-47c4-4401-8175-88d16e70e1b6" />